main.py

```diff
@@ -0,0 +1,24 @@
+import pandas
+import matplotlib.pyplot as plt
+
+data = pandas.read_csv('before_fix_raw.csv')
+
+data['minute'] = pandas.to_datetime(data['minute'])
+data['outage'] = (data['minute'] - data['minute'].shift(-1))
+
+res = data[data['outage'] > pandas.to_timedelta(1, unit='m')]
+outageTime = res['outage'].sum()
+
+totalTime = data['minute'].iloc[0] - data['minute'].iloc[-1]
```

Comment:
```suggestion
totalTime = endDate - startDate
```
should should pass start date and end date of your request, not calculate them
also you should pass period alis lime 'minute' or 'hour', not hardcode them (lines 6, 7)